Makefile

```diff
@@ -10,44 +10,56 @@
 	coverage \
 	coverage-html
 
-# Project constants
+# Build constants
 VERSION ?= 0.7
 PKGNAME ?= rhc-worker-script
 GO_SOURCES := $(wildcard src/*.go)
-PYTHON ?= python3
-PIP ?= pip3
-VENV ?= .venv3
-PRE_COMMIT ?= pre-commit
 GO_VERSION ?= 1.19
-CLIENT_ID ?= 00000000-0000-0000-0000-0000000000000
-SERVED_FILENAME ?= example_bash.yml
+
+BUILDFLAGS ?=
+LDFLAGS ?=
+ifeq ($(shell find . -name vendor), ./vendor)
+BUILDFLAGS += -mod=vendor
```

Comment:
Is this the main change that makes it work?

Reply:
Ah alright, with new build from this PR I am seeing now:

```
[root@vm-10-0-186-58 ~]# go version -m /usr/libexec/rhc/rhc-script-worker
/usr/libexec/rhc/rhc-script-worker: go1.20.12
	path	command-line-arguments
	dep	git.sr.ht/~spc/go-log	v0.1.1	
	dep	github.com/golang/protobuf	v1.5.3	
	dep	github.com/google/uuid	v1.6.0	
	dep	github.com/redhatinsights/yggdrasil	v0.0.0-20210811162724-41397343c25b	
	dep	golang.org/x/net	v0.23.0	
	dep	golang.org/x/sys	v0.18.0	
	dep	golang.org/x/text	v0.14.0	
	dep	google.golang.org/genproto/googleapis/rpc	v0.0.0-20230822172742-b8732ec3820d	
	dep	google.golang.org/grpc	v1.59.0	
	dep	google.golang.org/protobuf	v1.33.0	
	dep	gopkg.in/yaml.v3	v3.0.1	
	build	-buildmode=pie
	build	-compiler=gc
	build	-ldflags="-linkmode=external -compressdwarf=false -B 0x328b80c02072d97666206e224249b41795aebe27 -extldflags '-Wl,-z,relro '"
	build	CGO_ENABLED=1
	build	CGO_CFLAGS=
	build	CGO_CPPFLAGS="-D_FORTIFY_SOURCE=2 -fstack-protector-all"
	build	CGO_CXXFLAGS=
	build	CGO_LDFLAGS=
	build	GOARCH=amd64
	build	GOOS=linux
	build	GOAMD64=v1

```

Reply:
And this is how it looks like with the main build: 

```bash
[root@vm-10-0-184-120 ~]# scl enable go-toolset-1.19 'go version -m /usr/libexec/rhc/rhc-script-worker'
/usr/libexec/rhc/rhc-script-worker: go1.20.12
        path    rhc-script-worker-0.7
        build   -buildmode=pie
        build   -compiler=gc
        build   -ldflags=" -linkmode=external -B 0x0390eaffde65c3735698ff0a0bebbb62255fc0f4 -extldflags '-Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '"
        build   -tags=rpm_crashtraceback
        build   CGO_ENABLED=1
        build   CGO_CFLAGS=
        build   CGO_CPPFLAGS=
        build   CGO_CXXFLAGS=
        build   CGO_LDFLAGS=
        build   GOARCH=amd64
        build   GOOS=linux
        build   GOAMD64=v1
```